internal/connector/test/integration/features/connector-agent-api.feature

```diff
@@ -1193,6 +1193,41 @@ Feature: connector agent API
     }
     """
 
+    #-----------------------------------------------------------------------------------------------------------------
+    # Here we test the PATCH admin API. Only desired_state is available through admin PATCH
+    #-----------------------------------------------------------------------------------------------------------------
+    When I PATCH path "/v1/admin/kafka_connectors/${connector_id}" with json body:
+        """
+        {
+            "desired_state": "stopped"
+        }
+        """
+    Then the response code should be 202
+
+    When I GET path "/v1/admin/kafka_connectors/${connector_id}"
+    Then the response code should be 200
+    And the ".desired_state" selection from the response should match "stopped"
+
+    When I PATCH path "/v1/admin/kafka_connectors/${connector_id}" with json body:
+        """
+        {
+            "desired_state": "ready"
+        }
+        """
+    Then the response code should be 202
+
+    When I GET path "/v1/admin/kafka_connectors/${connector_id}"
+    Then the response code should be 200
+    And the ".desired_state" selection from the response should match "ready"
+
+    When I PATCH path "/v1/admin/kafka_connectors/${connector_id}" with json body:
```

Comment:
Instead of a generic non-existent field like `some_stuff`, the test should use an input json that would be valid with the public API with connector properties and kafka connection settings to test that only the `desired_state` can be patched. 
The test should also check the error response for correct error code and message.

Reply:
I don't agree this test is needed. Having `ConnectorAdminRequest` with only `desired_state` makes this scenario impossible. 

Currently this test aims to check for a error response when the request is malformed, which can be a valid scenario.

Reply:
Agreed, with the change in request type, we don't have to worry about unmodefiable fields.